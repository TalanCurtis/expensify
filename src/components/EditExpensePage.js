import React from 'react';

const EditExpensePage = (props) => (
  <div>
    EditExpensePage
    editing
  </div>
);

export default EditExpensePage;