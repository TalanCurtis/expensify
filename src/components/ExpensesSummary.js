import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
  const formatedExpensesTotal = numeral(expensesTotal).format('$0,0.00');
  
  return (
    <div>
        <h1>{`Viewing ${expensesCount} ${expenseWord} totaling ${formatedExpensesTotal}`}</h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: expensesTotal(visibleExpenses) / 100,
  };
};

export default connect(mapStateToProps)(ExpensesSummary);