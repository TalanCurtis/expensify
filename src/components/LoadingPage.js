import React from 'react';

const LoadingPage = (u) => (
  <div className="loader">
    <img className="loader__image" src="/images/loader.gif" alt=""/>
  </div>
);

export default LoadingPage;