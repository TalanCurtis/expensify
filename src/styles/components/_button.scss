.button {
  background-color: $light-blue;
  border: none;
  border-radius: 3px;
  color: white;
  display: inline-block;
  line-height: 1;
  padding: $s-size;
  font-size: $font-size-large;
  font-weight: 500;
  text-decoration: none;
}

.button--link {
  background: none;
}