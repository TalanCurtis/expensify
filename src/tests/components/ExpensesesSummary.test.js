import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary }  from '../../components/ExpensesSummary';

test('should correctly render ExpenseSummary with 1 expense', ()=>{
  const wrapper =  shallow(<ExpensesSummary  expensesCount={1} expensesTotal={234} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpenseSummary with multiple expense', ()=>{
  const wrapper =  shallow(<ExpensesSummary  expensesCount={33} expensesTotal={2343434} />);
  expect(wrapper).toMatchSnapshot();
});