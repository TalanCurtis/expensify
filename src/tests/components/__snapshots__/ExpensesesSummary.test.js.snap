// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 1 expense totaling $234.00
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 33 expenses totaling $2,343,434.00
  </h1>
</div>
`;
